clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/AbstractStorageEngine.java

```diff
@@ -356,7 +356,7 @@ public synchronized void close() throws VeniceException {
     tmpList.parallelStream().forEach(Partition::close);
     LOGGER.info(
         "Closing {} rockDB partitions of store {} took {} ms",
-        partitionList.size(),
```

Comment:
Nit: no difference with/without this change, right?

Reply:
there is, it prints 0 for spart list.

Reply:
I see.
So sparse list doesn't implement this method properly, so either we should reimplement that method or throw exception to claim it is not supported, otherwise, it is error prone.

Reply:
Actually sparseList.size would return size of array even if its sparsely populated. So if a host has partition 1 and 100. size will return 101.

Reply:
Got it.
It is still not properly implemented from my understanding.
@FelixGV 
Shall we re-implement it to return the actual element count? since array.size is more like an implementation detail, which shouldn't change the semantics of `size` function.

Reply:
Actually Felix implemented a function `nonNullSize` for this purpose.

Reply:
Shouldn't switch the semantics instead since `#size` is supposed to return `non-null` element num.

Reply:
For sparse list we can 2 types of size. 
1.  the underlying array's size (capacity) which includes both empty and non-empty slots.
2.  nonNullSize which Felix added. 
Are you asking if size should be same as nonNullSize?

Reply:
I don't know if there is an easy answer to this one... because as a list, I think it's reasonable for a user to expext having seen every element if they do this:

```
for (int i = 0; i < list.size(); i++) {
  list.get(i);
}
```

The above does indeed traverse all elements, as a user might reasonably expect. If we make `size` return the non null size, then the above traversal code is likely to short-circuit too soon (i.e. unless the list happens to be densely populated).

So... directionally, I think the clean thing to do is to evolve this SCL utility into a full-fledged Map, rather than a List. Then it could become less confusing. As of now, it is kind of in-between a Map and a List, and that causes confusion.

Reply:
The underlying array is an implementation detail and the `size` function is supposed to respect API contract defined by the interface from my understanding:
```
    /**
     * Returns the number of elements in this list.  If this list contains
     * more than <tt>Integer.MAX_VALUE</tt> elements, returns
     * <tt>Integer.MAX_VALUE</tt>.
     *
     * @return the number of elements in this list
     */
    int size();
```
I supposed the above `number of elements` are the ones we explicitly put in, right?

Reply:
I see your point. Currently `size` kinda returns the capacity of the array.  But developers could easily make mistakes if they want to iterate the array using for loop with size() and miss to visit all the elements. BTW can we separate the discussion from this PR ?

Reply:
Sure, that will be fine.
Please cut a ticket to track this since this API is error-prone from the user POV.